### PR TITLE
Add v1 API

### PR DESCRIPTION
**Pull request by damemi**

### CODE REVIEW

**Review by damemi — COMMENTED** [review]
/cc @soltysh @ingvagabund 
I think I've added the internal API correctly (but running the generate script locally will depend on https://github.com/kubernetes/kubernetes/pull/100666)

Next step is adding the code to actually consume/convert whatever version is passed as a CRD. This will let us bridge existing stored v1beta1 versions to the new v1 version without the extra fields

**Comment by damemi** [comment]
At this point it runs properly for me, but the CRD schema gen needs work to patch it. I just generated this with `controller-gen` manually from the latest upstream branch (rather than our makefile targets, which is fine since we don't need to just patch the openapi schema)

File: pkg/operator/target_config_reconciler.go [hunk 1]
```diff
@@ -3,6 +3,7 @@ package operator
 import (
 	"context"
 	"fmt"
+	deschedulerv1 "github.com/openshift/cluster-kube-descheduler-operator/pkg/apis/descheduler/v1"
```
Comment (ingvagabund) [inline pkg/operator/target_config_reconciler.go:6]: nit: move this next to other `github.com/openshift/cluster-kube-descheduler-operator` imports

File: pkg/apis/descheduler/v1/doc.go [hunk 2]
```diff
@@ -4,5 +4,4 @@
 // +groupName=operator.openshift.io
 // +groupGoName=Kubedeschedulers
 // +kubebuilder:validation:Optional
-// +k8s:conversion-gen=github.com/openshift/cluster-kube-descheduler-operator/pkg/apis/descheduler
```
Comment (ingvagabund) [inline pkg/apis/descheduler/v1/doc.go]: This was added in the first commit. Can you remove it there instead?

File: pkg/apis/descheduler/v1beta1/doc.go [hunk 3]
```diff
@@ -4,5 +4,4 @@
 // +groupName=operator.openshift.io
 // +groupGoName=Kubedeschedulers
 // +kubebuilder:validation:Optional
-// +k8s:conversion-gen=github.com/openshift/cluster-kube-descheduler-operator/pkg/apis/descheduler
```
Comment (ingvagabund) [inline pkg/apis/descheduler/v1beta1/doc.go]: This was added in the first commit. Can you remove it there instead?

**Review by ingvagabund — CHANGES_REQUESTED** [review]
Just nits

**Comment by damemi** [comment]
Thanks @ingvagabund, I reorganized some of the commits and removed the Makefile changes since those are broken.

So right now, this can't generate a new CRD if there are any changes. I don't think that is technically a blocker, since we have the ability to manually generate any CRD changes should they come up. But it is definitely an issue that could warrant its own bugzilla. @soltysh what do you think about this?

**Review by soltysh — APPROVED** [review]
/lgtm

**Pull request merged**